legacylogin/src/main/java/com/hamdan/forzenbook/legacy/login/LegacyLoginActivity.kt

```diff
@@ -0,0 +1,146 @@
+import com.hamdan.forzenbook.legacy.login.databinding.ActivityLegacyLoginBinding
+import com.hamdan.forzenbook.ui.core.R
+import dagger.hilt.android.AndroidEntryPoint
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.withContext
+
+@AndroidEntryPoint
+class LegacyLoginActivity : ComponentActivity() {
+    // TODO FA-104 add a click listener to send to Create Account on the createAccount text
+    private lateinit var binding: ActivityLegacyLoginBinding
+    private val loginModel: LegacyLoginViewModel by viewModels()
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        binding = ActivityLegacyLoginBinding.inflate(layoutInflater)
+        val view = binding.root
+        setContentView(view)
+        binding.apply {
+            var emailInputTextValue = ""
+            var codeInputTextValue = ""
+            loginSubmitButton.isEnabled = false
+            loginClickBlocker.setOnClickListener { }
+
+            emailErrorText.text = getString(R.string.login_email_error_format)
+            codeErrorText.text = getString(R.string.login_code_error)
+
+            loginSubmitButton.setOnClickListener {
+                loginModel.loginClicked()
+            }
+
+            loginModel.viewModelScope.launch(Dispatchers.IO) {
+                loginModel.state.collect { state ->
+                    withContext(Dispatchers.Main) {
+                        state.apply {
+
+                            emailInputTextValue = email.text
+                            codeInputTextValue = code.text
+
+                            loginSubmitButton.isEnabled = if (!inputtingCode) {
+                                email.error.isValid()
+                            } else {
+                                email.error.isValid() && code.error.isValid()
+                            }
+
+                            emailErrorText.apply {
+                                if (!email.error.isValid() && email.text.isNotEmpty()) {
+                                    if (email.error == LoginError.EmailError.Length) {
+                                        text = getString(R.string.login_email_error_length)
+                                        visibility = View.VISIBLE
+                                    } else if (email.error == LoginError.EmailError.InvalidFormat) {
+                                        text = getString(R.string.login_email_error_format)
+                                        visibility = View.VISIBLE
+                                    }
+                                } else {
+                                    visibility = View.GONE
+                                }
+                            }
+
+                            (if (inputtingCode) View.VISIBLE else View.GONE).apply {
+                                val visible = this
+                                inputCodeBody.visibility = visible
```

Comment:
Replace everywhere you use `.visibility` with `isVisible`

Reply:
replaced